src/cockpit/superuser.py

```diff
@@ -66,9 +73,6 @@ async def do_connect_transport(self) -> None:
             transport = await self.spawn(self.args, env, stderr=agent, start_new_session=True)
 
             if '# cockpit-bridge' in self.args:
```

Comment:
I suppose splitting this code into two parts was done due to some changed ordering requirements? Although it's not clear to me why writing the askpass helper above would depend on an already initialized InteractionAgent -- in fact, the old ordering made more sense to me, as the agent surely depends on a working askpass script?

This ordering change could use at least some justification in the commit message.

Testing for this magic `# cockpit-bridge` string twice is a DRY. Not a strong veto, but it may be nicer to set a boolean flag for this, if the code split really is necessary?

Reply:
It's because the `.add_handler()` call on the interaction agent disappeared.  All handlers have to be set at the start now, which necessitates the earlier creation of the handler.

I have an idea for how this can be done more elegantly.